synapseclient/entity.py

```diff
@@ -244,7 +244,7 @@ def __init__(self, properties=None, annotations=None, local_state=None, parent=N
                 # Re-map `items` to `datasetItems` to avoid namespace conflicts
                 # between Dataset schema and the items() builtin method.
                 if 'items' in properties:
-                    properties['datasetItems'] = properties['items']
+                    properties['dataset_items'] = properties['items']
```

Comment:
Is this a bug-fix with the current implementation?  We should add a test for this.